### ISSUE TITLE
Python3.6 '--no-luajit' encoding error

### ISSUE THREAD

**Issue by 5up3rD4n1** [scoder/lupa#100]
I'm trying to create a python3 compatible version for https://github.com/jbaiter/chdkptp.py, but a I read in https://docs.python.org/3/howto/pyporting.html#text-versus-binary-data, python 2 and 3 are not compatibles when using strings as byte types. As I can understand the issue here is that the python2 version had no problem using strings as bytestring all over the code. Now using python 3 if I create a `LuaRuntime` with no encoding I get an error while decoding the bytes buffer from lua and if I use `encoding=None` parameter I get a lot of messages like `attempting to use global (a userdata value)` which means that for Lua runtime the data can not be accessed. I tried to use `tostring` lua function everywhere but I don't know where I should used it or not.

 

```python3
Python 3.6.3 (default, Oct  3 2017, 21:45:48) 
[GCC 7.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import chdkptp
>>> device = chdkptp.ChdkDevice(chdkptp.list_devices()[0])
>>> device.shoot()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/dsolis/.virtualenvs/librescan/lib/python3.6/site-packages/chdkptp/device.py", line 460, in shoot
    return self._shoot_streaming(options, dng=kwargs.get('dng', False))
  File "/home/dsolis/.virtualenvs/librescan/lib/python3.6/site-packages/chdkptp/device.py", line 547, in _shoot_streaming
    """)(img_data)
  File "lupa/_lupa.pyx", line 541, in lupa._lupa._LuaObject.__call__
  File "lupa/_lupa.pyx", line 1288, in lupa._lupa.call_lua
  File "lupa/_lupa.pyx", line 1310, in lupa._lupa.execute_lua_call
  File "lupa/_lupa.pyx", line 1332, in lupa._lupa.unpack_lua_results
  File "lupa/_lupa.pyx", line 1099, in lupa._lupa.py_from_lua
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 0: invalid start byte
>>> 
```

I appreciate any help I can get or if the is some library I can use to decode the string from Lua to python. I also tried the native code posted here http://lua-users.org/wiki/LuaUnicode to use an utf-32 string, but didn't work, I believe it is related than the resulting string comes from an image and then a lot of unicode can not be parsed. I can provide more information if there is something missing. 

I'm taggain @jbaiter just if he have an idea of how to deal with this. 

Basically what I need is the bytecodes of the picture captured by Lua. But I guess lupa is trying to decode it. Maybe an option to eval code overwriting the default encoding and return the bytestring can do it. 

Thanks in advance.

**Comment by TitanSnow** [comment]
Maybe you can try `encoding='latin_1'` and then decode the str using `latin_1`
